### DISCUSSION TITLE
Regarding CPU usage rate

### DISCUSSION THREAD

**Discussion by chenchuanliang** [oracle/opengrok#4658] [Q&A]
My computer's service CPU has been running at over 90%. The previously installed 22.04 computer did not have this issue, but the same computer also experienced the same problem after reinstalling 18.04.
as follows:
![17254136499977](https://github.com/user-attachments/assets/99804238-f69c-4b60-9e7a-56494ef59b94)
Environment:
OpenGrok 1.7.39 、1.13.17
OpenJdk  17
Ubuntu 18.04  5.4.0-131-generic  i7-6700 CPU @ 3.40GHz
Is it normal for CPU utilization to be so high？
Thank you

**Comment by vladak** [comment]
It depends on what is this java process. If this is the indexer then it depends on what it is doing at the moment. The indexer is heavily parallelized so during the indexing it is quite common to consume all available resources if there is enough workload, unless it is [tuned](https://github.com/oracle/opengrok/wiki/Indexer-configuration#indexer-tunables) down by using the various thread/memory limits.

In general you can get more glimpse into the indexer operations w.r.t. the resources by:
  - bump log level and see the log entries added to the log while the CPU usage is high
  - getting several thread stack snapshots using `jstack`
  - ideally also setup the [monitoring](https://github.com/oracle/opengrok/wiki/Monitoring)
